### ISSUE TITLE
feat: apr finetune — LoRA/QLoRA fine-tuning with adapter upload to Hub

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#244]
## Summary

Upgrade `apr tune` from a planning-only tool to a full fine-tuning pipeline that executes LoRA/QLoRA training via entrenar. This is the **#2 most common operation** on HuggingFace models — PEFT/LoRA is the standard for parameter-efficient fine-tuning, and users expect to fine-tune then re-upload adapters (or merged models) to the Hub.

Currently `apr tune` only plans configurations and estimates VRAM. It cannot execute training.

## Motivation

- **Hub ecosystem**: LoRA adapters are the second most uploaded derivative. PEFT's `adapter_config.json` + weights format is the standard.
- **entrenar already has the building blocks**: `entrenar::lora` has adapter, layer, config, gradient_tests, and QLoRA support. `entrenar::finetune` has the training loop.
- **Gap**: `apr tune --train-data data.jsonl` is accepted as a CLI arg but does not execute training. No adapter export or Hub upload.

## Proposed CLI

```bash
# LoRA fine-tuning (rank auto-selected)
apr finetune model.apr --method lora --data train.jsonl -o adapter/

# QLoRA (4-bit base + fp16 adapters)
apr finetune model.apr --method qlora --data train.jsonl --rank 16 -o adapter/

# Full fine-tuning (unfreeze all)
apr finetune model.apr --method full --data train.jsonl -o finetuned.apr

# Merge adapter back into base
apr finetune merge model.apr --adapter adapter/ -o merged.apr

# Upload adapter to Hub
apr finetune push adapter/ --repo user/my-model-lora

# Planning mode (VRAM/time estimate)
apr finetune model.apr --method qlora --plan --json
```

Note: Consider renaming `apr tune` → `apr finetune` for clarity, or keeping `tune` as alias.

## Implementation Path

1. Wire `entrenar::lora` and `entrenar::finetune` into aprender CLI
2. Add training loop execution (currently only planning)
3. Add adapter export in PEFT-compatible format (`adapter_config.json` + `adapter_model.safetensors`)
4. Add adapter merge (`entrenar::lora::adapter` merge functionality)
5. Add `apr finetune push` via `entrenar::hf_pipeline` or direct Hub API
6. JSON progress output for CI integration

## Entrenar Modules to Surface

| Module | Purpose |
|--------|---------|
| `entrenar::lora::adapter` | LoRA adapter creation and management |
| `entrenar::lora::layer` | LoRA layer injection |
| `entrenar::lora::qlora` | QLoRA 4-bit base loading |
| `entrenar::lora::config` | LoRA hyperparameter configuration |
| `entrenar::finetune` | Training loop, optimizer, scheduler |
| `entrenar::autograd` | Backward pass for training |

## Acceptance Criteria

- [ ] `apr finetune model.apr --method lora --data train.jsonl` trains and saves adapter
- [ ] `apr finetune merge` produces valid APR model
- [ ] Adapter format compatible with PEFT (`adapter_config.json`)
- [ ] `--plan --json` returns VRAM/time estimates (existing behavior preserved)
- [ ] Testable with tiny models (SmolLM-135M fine-tune should complete in <5min)

## References

- [How to fine-tune open LLMs in 2025](https://www.philschmid.de/fine-tune-llms-in-2025)
- [PEFT LoRA docs](https://huggingface.co/docs/peft/main/en/developer_guides/lora)
- [Unsloth (2x faster LoRA)](https://unsloth.ai)
- Existing issue: #176 (closed — planning-only implementation)
- entrenar source: `src/lora/`, `src/finetune/`, `crates/entrenar-lora/`

**Comment by noahgift** [comment]
Implemented apr finetune with LoRA/QLoRA planning via entrenar-lora, memory breakdown, adapter merge. 8+ tests, clippy clean.
